Implemented metastore summary

vikrantpuppala opened the pull request:
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users. -->

SemyonSinchenko committed:
parent 5b917869a2587d26df5c477e6af401385371661a
author Semyon <ssinchenko@apache.org> 1725664890 +0200
committer Vikrant Puppala <vikrant.puppala@databricks.com> 1728321729 +0530
gpgsig -----BEGIN PGP SIGNATURE-----

 iQIzBAABCAAdFiEEuLQK5O9DgM4B7E4mMF6GPKJC6IkFAmcEGMEACgkQMF6GPKJC
 6Ilmbw/+OZA3QTpO1bd5heNntXgyyE6x9LcuIQO05gY/K2egAv7nyB7sm+pPgk0Y
 /AaqheHCrQCQSVUCY6xNVp288mCIRQYHOCPfPU2JfeLXxug+9FzNzeT/mI1thXXf
 lkf7ae5sU/m853putylWBTbyBwr1VU+eUm5n8yzggVyKYCgaoP6kgOcX2EcATJ65
 KYj/L+97kzJDe4Kj9W3/MiaOzHsiHShBGRsE4okqRQCi5zd26hmWw13rAVWceYA7
 2BHl8zVr+maHYz2p9897ELarzbMo9JGiSykvJbpj+jPC3RIZdh1agXlPEKTCeOQa
 1qho44jxmdL1Vv1vEv3B81YHem3CkM5n+1xyY9fFQHrhA/3DU+TO3oQLwz7aeN80
 kOII62ogGtR0oOf/foLmMy71s4yP26aBg92Zfkhw8RSrKGZH+IEL8GZGVdCnLbwZ
 NgTvcCC8ySMMs7dZ7y5TVsHe/Qf0YXSivmiFMu9AqsmfMB4Ig9wbfCvSrHA8gX+P
 z1FDCzSCuXvpgSvGJAIsHt5v85qDbeQbflr2YYhQBdzz1MZyOImZc6M4mp6s+KpH
 fv+Z4WFeqqzdjddePPVYG49s2g4QLFmZ+LY3yITlaA5R0HwFMH0vwsi0+jvUujbN
 mupC4tLB8/TMVAeskpJfmA46flEtdQGLVHxnKvjKvpRVDNfVS40=
 =imRP
 -----END PGP SIGNATURE-----

impl for metastore summary

Fix javadoc generation issue (#519)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Fix javadoc generation issue

e.g. `unknown tag: AuthorizeKey`

Include UI and auth docs (#528)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
Including auth and UI docs. Slight restructuring of docs hierarchy.
Clean up missing link, add cross linking, etc.

<!-- Please state what you've changed and how it might affect the users.
-->

---------

Co-authored-by: Jamie Knight <jamie.knight@databricks.com>
Co-authored-by: Shawn Chen <ychshawn@gmail.com>
Co-authored-by: Mocker <1179360+creechy@users.noreply.github.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

Feat/delete model version (#517)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Ability to delete a model version from the UI.
<img width="425" alt="Screenshot 2024-09-26 at 2 29 57 PM"
src="https://github.com/user-attachments/assets/fb27e047-f642-4221-8ef2-3b5ea96091a5">
<img width="522" alt="Screenshot 2024-09-26 at 2 29 51 PM"
src="https://github.com/user-attachments/assets/d3b83207-3b21-44c5-8089-04d5370c3118">
<img width="226" alt="Screenshot 2024-09-26 at 2 29 41 PM"
src="https://github.com/user-attachments/assets/b4726f0e-b5ad-4ac5-a1b6-77c9a3aba815">

Update docs to include CLI managing users (#530)

**PR Checklist**

- [X] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
- Fixed image loc in UI doc
- Added managing users to CLI doc

<!-- Please state what you've changed and how it might affect the users.
-->

Update auth.md (#531)

**PR Checklist**

- [X] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
Clarification of Identity providers supported

<!-- Please state what you've changed and how it might affect the users.
-->

Signed-off-by: Denny Lee <denny.g.lee@gmail.com>

revert unknown changes

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

comments

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

add authorizer

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

fmt

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

sumeet-db commented on server/src/main/java/io/unitycatalog/server/persist/MetastoreRepository.java:
```diff
@@ -1,19 +1,75 @@
+  public GetMetastoreSummaryResponse getMetastoreSummary() {
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      MetastoreDAO metastoreDAO = getMetastoreDAO(session);
```
If metastoreDAO never changes after init, should we cache the response of `initMetastoreIfNeeded` and use that instead?

sumeet-db commented on server/src/main/java/io/unitycatalog/server/utils/Constants.java:
```diff
@@ -7,6 +7,7 @@ public class Constants {
   public static final String SCHEMA = "schema";
   public static final String TABLE = "table";
   public static final String FUNCTION = "function";
+  public static final String DEFAULT_METASTORE_NAME = "uc_oss_metastore";
```
Wondering if this should be user-settable with some default. The default can contain ID (with a better prefix) to uniquify across server instances.

sumeet-db replied:
@tdas Will need your expertise here.

kriscon-db committed:
Make source required (#566)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Source was erroneously left off of the required list for
CreateModelVersion when checked in. In this PR we:

- Add source as required at at the API
- Add service handler checks for source existence
- Add an explicit test in the unit test to validate that source must be
filled in.

jaceklaskowski committed:
[MINOR] Remove duplicate + no publics in interface (#561)

This is a follow-up to f9a9bf1f84cb4c7fa03ecb569ae19f306a7cd85b:

1. Removes a code duplication
2. `@Get("/function/{name}")`
3. Removes `public` qualifiers in `UnityCatalogAuthorizer` interface

/cc @creechy

Signed-off-by: Jacek Laskowski <jacek@japila.pl>

jamieknight-db committed:
[UI] Feat/update model version description (#543)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

User can update model version description via the UI. 

<img width="422" alt="Screenshot 2024-10-01 at 5 07 00 PM"
src="https://github.com/user-attachments/assets/72519f61-0105-42b9-bea2-71987bfebce2">
<img width="440" alt="Screenshot 2024-10-01 at 5 06 40 PM"
src="https://github.com/user-attachments/assets/e8edd838-8871-473b-be45-00bf47cf04db">
<img width="523" alt="Screenshot 2024-10-01 at 5 06 35 PM"
src="https://github.com/user-attachments/assets/edfaf5e6-bd26-4e53-8e34-c9eb46ec3d83">
<img width="379" alt="Screenshot 2024-10-01 at 5 06 12 PM"
src="https://github.com/user-attachments/assets/7349da70-cc8e-447b-acd3-3c205ab7fd02">

jamieknight-db committed:
[UI] Edit registered model description (#562)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

User can add/edit model description via the UI.

<img width="532" alt="Screenshot 2024-10-03 at 1 28 41 PM"
src="https://github.com/user-attachments/assets/ae2222ec-f426-4cd4-96b9-ab41f317c985">
<img width="261" alt="Screenshot 2024-10-03 at 1 28 14 PM"
src="https://github.com/user-attachments/assets/d2b5d701-6b1e-4075-a3df-d7d4fbc95851">
<img width="411" alt="Screenshot 2024-10-03 at 1 28 06 PM"
src="https://github.com/user-attachments/assets/39329596-272f-4469-9533-ff7aa595a988">
<img width="516" alt="Screenshot 2024-10-03 at 1 28 01 PM"
src="https://github.com/user-attachments/assets/2170fa3b-b1a6-4d57-af7a-9d6e83f7bd2d">
<img width="355" alt="Screenshot 2024-10-03 at 1 27 47 PM"
src="https://github.com/user-attachments/assets/3bb567cc-020e-4aa5-a2b9-695b744a7a8f">

sudharshanraja-db committed:
Add patch support for scim user endpoint (#500)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
 - Added patch endpoint  to handle user removal
 - Marks the user status as `disabled` in uc db when removed
 - Reenable  the user status to enable when the user is added back
 
Attaching the screenshots for all the scenarios
<img width="1066" alt="add_user_to_app"
src="https://github.com/user-attachments/assets/50cf1cd5-1e28-4e1c-9432-4ccf78fdd34e">

[add_user_uc_db.pdf](https://github.com/user-attachments/files/17093491/add_user_uc_db.pdf)


This PR closes #423  

<img width="1027" alt="unassign_user_from_app"
src="https://github.com/user-attachments/assets/9bdeb2ac-6b80-4607-9e3a-13e473ba1517">
<img width="1011" alt="user_add_back_to_app"
src="https://github.com/user-attachments/assets/0af56fe9-dc75-41d2-b24b-4f60fc2d39f7">

[user_disabled_in_uc_Db.pdf](https://github.com/user-attachments/files/17093493/user_disabled_in_uc_Db.pdf)

[user_status_enabled_bak_in_db.pdf](https://github.com/user-attachments/files/17093494/user_status_enabled_bak_in_db.pdf)

---------

Signed-off-by: sudharshanraja-db <102200109+sudharshanraja-db@users.noreply.github.com>
Co-authored-by: Denny Lee <denny.g.lee@gmail.com>

vikrantpuppala committed:
Merge branch 'metastore-api' into metastore-impl

# Conflicts:
#	api/Apis/MetastoresApi.md
#	api/Models/UpdateModelVersion.md
#	api/Models/UpdateRegisteredModel.md
#	api/README.md
#	api/all.yaml
#	connectors/spark/src/test/java/io/unitycatalog/spark/TableReadWriteTest.java
#	docs/quickstart.md
#	server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java
#	server/src/main/java/io/unitycatalog/server/persist/MetastoreRepository.java
#	server/src/main/java/io/unitycatalog/server/service/ModelService.java
#	ui/src/hooks/models.tsx
#	ui/src/pages/ModelDetails.tsx

vikrantpuppala committed:
fix

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
remove java version

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
Ensure that docs are always updated using `build/sbt generate` (#535)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users.
-->
Add github action to ensure that docs are always updated using
`build/sbt generate`

---------

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

serena-ruan committed:
ucai-core lib with base UC function client (#536)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
Based on https://github.com/unitycatalog/unitycatalog/issues/525:
- Implement base UC function client


<!-- Please state what you've changed and how it might affect the users.
-->

---------

Signed-off-by: serena-ruan_data <serena.ruan@databricks.com>
Signed-off-by: Denny Lee <denny.g.lee@gmail.com>
Co-authored-by: Denny Lee <denny.g.lee@gmail.com>
Co-authored-by: Ben Wilson <39283302+BenWilson2@users.noreply.github.com>

jaceklaskowski committed:
Use one version of Delta Lake consistently (#541)

This change makes sure that there is one version of Delta Lake (based on
a lazy val shared across subprojects).

Signed-off-by: Jacek Laskowski <jacek@japila.pl>

vikrantpuppala committed:
cache metastore DAO

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
Merge remote-tracking branch 'unitycatalog/main' into metastore-impl

# Conflicts:
#	.gitignore

vikrantpuppala committed:
cache metastore DAO

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
fix tests

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
Merge branch 'metastore-api' into metastore-impl

vikrantpuppala committed:
only keep metastore id in resp

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
Merge branch 'metastore-api' into metastore-impl

serena-ruan committed:
ucai-core databricks utils (#537)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
FR: https://github.com/unitycatalog/unitycatalog/issues/525
Add some utility functions for validating params in preparation for
databricks client integration.

<!-- Please state what you've changed and how it might affect the users.
-->

---------

Signed-off-by: serena-ruan_data <serena.ruan@databricks.com>

serena-ruan committed:
Core tests (#538)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
Add tests for core UC function client.

<!-- Please state what you've changed and how it might affect the users.
-->

---------

Signed-off-by: serena-ruan <serena.rxy@gmail.com>

vikrantpuppala committed:
comments

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

serena-ruan committed:
databricks function client support (#578)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
Databricks function client support, implement all methods in the
BaseFunctionClient and added unit tests + integration tests.
This PR is large mostly because of the test suite :)

<!-- Please state what you've changed and how it might affect the users.
-->

Signed-off-by: serena-ruan <serena.rxy@gmail.com>

jamieknight-db committed:
Feat/create model (#576)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

create registered model from the ui
<img width="1298" alt="Screenshot 2024-10-08 at 3 14 06 PM"
src="https://github.com/user-attachments/assets/71c9989e-934a-49fa-948b-2e7102b83c5c">
<img width="863" alt="Screenshot 2024-10-08 at 3 13 54 PM"
src="https://github.com/user-attachments/assets/8ebbfbf0-2d9b-4d77-8d43-7ea38d2bd775">
<img width="414" alt="Screenshot 2024-10-08 at 3 13 44 PM"
src="https://github.com/user-attachments/assets/06e9be5e-a256-4ded-a014-a206c3812c65">
<img width="523" alt="Screenshot 2024-10-08 at 3 12 25 PM"
src="https://github.com/user-attachments/assets/23b79ffd-b09e-4bd9-ba01-6ef660a2aa9c">
<img width="244" alt="Screenshot 2024-10-08 at 3 11 58 PM"
src="https://github.com/user-attachments/assets/793bcf75-e3db-4e8a-a479-ad598b5897ac">

jaceklaskowski committed:
Refactor ServerPropertiesUtils (to be ServerProperties) (#575)

As agreed with @alexreid-db, `ServerPropertiesUtils` should really be in
`io.unitycatalog.server.utils` package.

The other changes are mainly as follows:

1. No need for `Utils` suffix
2. Added `isAuthorizationEnabled` method
3. Javadoc

---------

Signed-off-by: Jacek Laskowski <jacek@japila.pl>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

ebyhr committed:
Add icon file for IntelliJ IDEA (#584)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

![Screenshot 2024-10-10 at 6 21
02](https://github.com/user-attachments/assets/80cc6b59-5a17-4b26-9076-8420b97287c9)

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>

ebyhr committed:
Remove redundant code (#586)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Remove redundant code

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

ebyhr committed:
Remove redundant TestUtils.toList (#585)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Remove redundant TestUtils.toList.

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

BenWilson2 committed:
[UCAI] Add callable support for python function creation (#577)

**PR Checklist**

- [X] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

This PR introduces the ability to provide a Google PyDoc formatted
python function with type hints to the client's `create_python_function`
API to generate a UC function from a base callable.

Note: generic types, Union types, class methods, instance methods, and
generic collection types are not permitted for registry due to the need
for strong typing within UC.

---------

Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

vksx committed:
Fix the page token bug for all the entities. (#375)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

- Fixed the page token bug for all the entities - Catalog, Schema,
Table, Volume, Function and Model. Now passing the page token will list
the the entities properly.

  Example:
  ```
  bin/uc catalog create --name test
  bin/uc catalog list --page_token test
  ```

- `listVolumes()` in the VolumeRepository now utilizes
PagedListingHelper to list and paginate the volumes.

Fixes #372

Signed-off-by: Vikas Sharma <master2vikas@gmail.com>

yaooqinn committed:
Add h/help option for unity catalog server (#504)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

This PR adds `h/help` option for `bin/start-uc-server` to print the
usage guide

```
 bin/start-uc-server -v
0.2.0-SNAPSHOT
  bin/start-uc-server -h
usage: bin/start-uc-server
 -h,--help         Print help message.
 -p,--port <arg>   Port number to run the server on. Default is 8080.
 -v,--version      Display the version of the Unity Catalog server
```

---------

Signed-off-by: Kent Yao <yao@apache.org>

ebyhr committed:
Remove unused method from TestUtils (#582)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Remove unused method from TestUtils

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

HorizonNet committed:
Fix broken link to Spark Connector in PuppyGraph integration (#568)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add
tests!
- [x] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Fix the broken link in the docs pointing to the Spark Integration
instead of the Spark Connector in the PuppyGraph integration
documentation page.

Resolves #564

Signed-off-by: Jan Hentschel <jan.hentschel@bricksaffinity.com>

yaoderek committed:
Docs: improve functions.md with supported language and notes on defining multi-line functions. (#499)

improve functions.md with supported language and notes on defining
multi-line functions.

**PR Checklist**

- [ X] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**
improve functions.md with supported language and notes on defining
multi-line functions.

HorizonNet committed:
Update Delta Lake table page to be closer to Formats (#552)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Resolves #546

Signed-off-by: Jan Hentschel <jan.hentschel@bricksaffinity.com>

avriiil committed:
docs: fix typos in Auth page (#558)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users.
-->

Signed-off-by: Avril Aysha <68642378+avriiil@users.noreply.github.com>
Co-authored-by: Denny Lee <denny.g.lee@gmail.com>

vikrantpuppala committed:
Merge branch 'metastore-api' into metastore-impl

vikrantpuppala committed:
remove changes to git ignore

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
Merge remote-tracking branch 'unitycatalog/main' into metastore-impl

# Conflicts:
#	server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

vikrantpuppala committed:
Add metastore summary API to the spec (#494)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users.
-->
This PR adds the metastore summary API to the spec. The implementation
will be added by follow-up PRs

---------

Signed-off-by: Denny Lee <denny.g.lee@gmail.com>
Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>
Signed-off-by: Vikrant Puppala <vikrantpuppala@gmail.com>
Co-authored-by: Ramesh Chandra <ramesh.chandra@gmail.com>

ebyhr committed:
Migrate JUnit assertions to AssertJ (#587)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Migrate JUnit assertions to AssertJ.

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>
Co-authored-by: Tathagata Das <tathagata.das1565@gmail.com>

ebyhr committed:
Use FileUtils.deleteDirectory (#591)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Use FileUtils.deleteDirectory

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>

ebyhr committed:
Remove unused ThrowingFunction (#592)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Remove unused ThrowingFunction

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>

vikrantpuppala committed:
Merge remote-tracking branch 'unitycatalog/main' into metastore-impl

vikrantpuppala committed:
format files and add java format check (#595)

**PR Checklist**

- [ ] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users.
-->
add java format check

---------

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

ebyhr committed:
Replace string concat with logger template (#590)

**PR Checklist**

- [x] A description of the changes is added to the description of this
PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add
tests!
- [ ] If you've added or modified a feature, documentation in `docs` is
updated

**Description of changes**

Replace string concat with logger template

Signed-off-by: Yuya Ebihara <ebyhry@gmail.com>

LuciferYang committed:
Reformat `ModelRepository.java` to pass javafmtCheck (#596)

https://github.com/unitycatalog/unitycatalog/pull/595 added
javafmtCheck, but the `ModelRepository.java` file submitted by
https://github.com/unitycatalog/unitycatalog/pull/590 was not formatted,
which resulted in a GA check error.

-
https://github.com/unitycatalog/unitycatalog/actions/runs/11300626359/job/31433764692

```
[warn] server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java isn't formatted properly!
[error] 1 file must be formatted
[error] (server / Compile / javafmtCheck) 1 file must be formatted
[error] Total time: 5 s, completed Oct 12, 2024, 8:11:48 AM
```

This pr used `build/sbt javafmt` to format the `ModelRepository.java`
file in order to fix the GA(javafmtCheck) issue.

Signed-off-by: yangjie01 <yangjie01@baidu.com>

vikrantpuppala committed:
Merge remote-tracking branch 'unitycatalog/main' into metastore-impl

# Conflicts:
#	server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

vikrantpuppala committed:
add tests

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>

vikrantpuppala committed:
test for server restart

Signed-off-by: Vikrant Puppala <vikrant.puppala@databricks.com>
